codebase2/codebase-sqlite/U/Codebase/Sqlite/Operations.hs

```diff
@@ -979,13 +979,16 @@ declReferentsByPrefix b32prefix pos cid = do
       h <- Q.expectPrimaryHashByObjectId oId
       let test :: ConstructorId -> Bool
           test = maybe (const True) (==) cid
-          cids = [cid | cid <- [0 :: ConstructorId .. ctorCount - 1], test cid]
+          cids =
+            if ctorCount == 0
+              then []
+              else [cid | cid <- [0 :: ConstructorId .. fromIntegral ctorCount - 1], test cid]
       pure (h, pos, dt, cids)
```

Comment:
Let's get rid of the `- 1` altogether; though I see that it is hairy due to the `Maybe ConstructorId`.
```suggestion
          cids = case cid of
            Nothing -> take ctorCount [0 :: ConstructorId ..]
            Just cid -> if fromIntegral cid < ctorCount then [cid] else []
      pure (h, pos, dt, cids)
```

Reply:
cool, I applied this suggestion